### PR TITLE
vdk-jupyter: Introduce unit tests for server request handlers

### PR DESCRIPTION
**Pull request by duyguHsnHsn**
What:
Introduced tests for currently existing handlers.

Why: 
Only manual tests were done so far. 

Signed-off-by: Duygu Hasan [hduygu@vmware.com](mailto:hduygu@vmware.com)

### CODE REVIEW

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py [hunk 1]
```diff
@@ -3,13 +3,98 @@
...
+    payload = json.loads(response.body)
+    assert payload == {
+        "error": "true",
+        "message": "¯\\_(ツ)_/¯\n"
+        "\n"
+        "what: Control Service Error\n"
+        "why: No host specified.\n"
+        "consequences: Operation cannot complete.\n"
+        "countermeasures: Verify that the provided url (--rest-api-url) is "
+        "valid and points to the correct host.\n",
```
Comment (ivakoleva) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py:45]: what's the expectation for the payload contents?

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py [hunk 2]
```diff
@@ -3,13 +3,98 @@
...
+    }
+
+
+async def test_create_job_post(jp_fetch):
```
Comment (ivakoleva) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py:71]: the purpose of the test is to reproduce a scenario where an empty data job name is provided for creation? if yes, please rename the method, to add information about the scenario it is covering

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py [hunk 3]
```diff
@@ -3,13 +3,98 @@
...
+        "\n"
+        "what: Control Service Error\n"
+        "why: No host specified.\n"
+        "consequences: Operation cannot complete.\n"
+        "countermeasures: Verify that the provided url (--rest-api-url) is "
+        "valid and points to the correct host.\n",
+    }
+
+
+async def test_download_job_post(jp_fetch):
```
Comment (ivakoleva) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py:49]: test rest api url empty ?

Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py:49]: I am testing only the handlers, not the operations. So I would say it doesn't matter whether it is a wrong input or not. It just returns an error message, but is still 200

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py (hunk 2 above)
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/tests/test_handlers.py:71]: I am not really testing the operations, but the handlers. I can provide tests for the operations too. I haven't done it bc the classes I am using are tested here: https://github.com/vmware/versatile-data-kit/tree/main/projects/vdk-control-cli/tests/vdk/internal/control/command_groups/job

**Review by ivakoleva — COMMENTED** [review]
I'm looking at the [gitlab pipeline](https://gitlab.com/vmware-analytics/versatile-data-kit/-/pipelines/768589723) for the tests, are we running them in the ci/cd?

**Pull request closed without merging**